### ISSUE TITLE
Motion Event Inspector window does not update itself with changes until the modified motion event is deselected

### ISSUE THREAD

**Issue by LB-MateuszSzweda** [o3de/o3de#12665]
**Describe the bug**
Any changes provided to the selected **motion event** by using its **Inspector** window will not update the window and the event itself until the modified motion event or motion is deselected and selected again in **Animation Editor** .

Please note that in order to update the **Inspector** window with new _Event Data_ brackets only the **motion event** has to be deselected and selected again but to update event _Start_ and _End time_ modifier the whole motion has to be deselected and selected again in the **Motion Set** window.

For further information please refer to the attached video file.

**Prerequisites**

**EMotion FX Animation** Gem is enabled and built for the active project.

**Steps to reproduce**

1. Launch the Editor and create a new level.
2. Open the **Animation Editor** tool.
3. Change its layout to _Animgraph_ and navigate to the _Motion Sets_ tab.
4. Add any motion to the **Motion Set** window (for instance _rin_Idle.fbx_).
5. Select the added motion and add a new event track in its **Time View** window.
6. Right-click the track and select _Add motion event_ option.
7. Click the added motion event and navigate to the **Inspector** window.
8. Add **TwoStringEventData** or **EventDataFootIK** to _Event Data_ bracket.
9. Modify _Start_ and _End time_ modifiers.
10. Deselect the Motion Event or the Motion itself and select it again.

**Expected behavior**
Motion Event Inspector window updates itself and the Motion event when changes are introduced to it.

**Actual behavior**
Motion Event Inspector window and the Motion itself are updated only after the Inspector window or motion is deselected and selected again.

**Screenshots/Video**

https://user-images.githubusercontent.com/86953456/196435390-15423fcd-052e-48de-9f88-ab22dcb3f576.mp4

**Found in Branch**
Development (b47b11d)

**Desktop/Device:**
- Device: PC
- OS: Windows
- Version: 10 Pro
- CPU: AMD Ryzen 9 3900X
- GPU: NVIDIA GeForce RTX 2060 SUPER
- Memory: 16GB

**Comment by hultonha** [comment]
This should now be fixed @LB-MateuszSzweda (see commit https://github.com/o3de/o3de/commit/ccba544b5f9b4e581012c6c025fede384b849faf)

**Comment by LB-MateuszSzweda** [comment]
Unfortunately the issue persisted for me even after I've merged Development (92af95a) with Development (ccba544) via Git fetch/Git merge actions and rebuilding the engine afterwards.
I've got the same results the motion even was not updating itself right after changing the values.

Just to be sure before reopening I'll try to reproduce it with a freshly cloned instance of O3DE  with the newest (9910a1e) commit.

**Comment by hultonha** [comment]
Thanks for checking @LB-MateuszSzweda, hmm this was definitely working for me (see video in PR). Can you try with latest development tomorrow and let me know. Thanks!

**Comment by LB-MateuszSzweda** [comment]
On todays Development (a881574) and (9910a1e)  the issue is indeed fixed and Motion Event Inspector correctly updates right away with given input and Event Data, however I've noticed that changing _Start_ and _End time_ modifiers is broken the other way around.

Changing them does not update the **Time View** window right away. Only after the motion is reselected or an _Event Data_ is modified the change is introduced to the **Time View**.
Which can be seen on the Video below.


https://user-images.githubusercontent.com/86953456/204517186-b29675a4-2ba9-4931-a120-d0e9964281cf.mp4


As this is a very similiar behavior this ticket can be reopened or I can file in a new one.

**Comment by hultonha** [comment]
Thanks for the update @LB-MateuszSzweda, hmm I'd create a separate ticket with the issue please. I take it this was also happening before correct? This wasn't something cause by the most recent change right? Cheers!

**Comment by LB-MateuszSzweda** [comment]
@hultonha Yes, It was present before and was not caused by recent change. I thought it was caused by the same bug as this ticker but after the fix it clarified as a separate issue.

I've logged it as https://github.com/o3de/o3de/issues/13458 .
And the above can be treated as Verified as fixed in Development (ccba544).

**Comment by hultonha** [comment]
Okay excellent thanks for the info @LB-MateuszSzweda 👍
